src/test/java/org/cactoos/map/MapEnvelopeTest.java

```diff
@@ -313,4 +314,17 @@ public void hashCodeFailsOnNull() {
             new MapEntry<>("key11", null);
         new MapOf<String, String>(first, second).hashCode();
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void mapEnvelopeShouldCompareDerivedClasses() {
```

Comment:
@scratchboom Please, test if an empty `MapEnvelope` and an empty derived are equals.

Reply:
@fabriciofx see new changes